app/models/automate_workspace.rb

```diff
@@ -0,0 +1,17 @@
+class AutomateWorkspace < ApplicationRecord
+  include UuidMixin
+  belongs_to :user
+  belongs_to :tenant
+  validates :tenant, :presence => true
+  validates :user, :presence => true
+
+  def output=(hash)
+    if hash['workspace'].blank? && hash['state_var'].blank?
+      raise ArgumentError, "No workspace or state_var specified for edit"
+    end
+
+    self[:output] = (output || {}).deep_merge(hash)
+    save!
```

Comment:
A setter should never call .save! directly.  It's should be the caller's concern as they may choose to defer saving, or use update_attributes or whatever.

Reply:
That being said, can the setter just be the setter, and deal with the validation at validation time?  Doing all this on a setter seems wrong.

Reply:
If you *must* have a method that sets and saves, then prefer creating a method like `def merge_output!(hash)` which is explicit in what it's doing